sycl/include/CL/sycl/handler.hpp

```diff
@@ -796,19 +796,11 @@ class __SYCL_EXPORT handler {
 
     // Get the kernal name to check condition 3.
     std::string KName = typeid(NameT *).name();
-    using KI = detail::KernelInfo<KernelName>;
-    bool DisableRounding =
-        (getenv("SYCL_DISABLE_PARALLEL_FOR_RANGE_ROUNDING") != nullptr) ||
-        (KName.find("SYCL_DISABLE_PARALLEL_FOR_RANGE_ROUNDING") !=
-         std::string::npos) ||
-        (KI::getName() == nullptr || KI::getName()[0] == '\0') ||
-        (KI::callsThisItem());
```

Comment:
I'm concerned about this particular change. Not calling `this_item` free functions is a requirement for range rounding feature. Having all tests passed means that we don't have coverage for this.

@romanovvlad, @rdeodhar, what do you think?

Reply:
The test SYCL/Basic/free_function_queries/free_function_queries.cpp tested this. However, for performance reasons, the heuristics for range rounding were later adjusted so that rounding is only done when the range  >= 1024. The test uses a range of 10 so the kernel that uses this_item is not subject to range rounding and the test passes.
The test will need to be modified to increase the range (I will do it separately).
So removing that check from handler.hpp is not correct.